src/compile/repeater.ts

```diff
@@ -20,40 +30,70 @@ export function replaceRepeaterInEncoding(encoding: Encoding<Field>, repeater: R
 /**
  * Replace repeater values in a field def with the concrete field name.
  */
-function replaceRepeaterInFieldDef(fieldDef: FieldDef<Field>, repeater: RepeaterValue): FieldDef<string> | null {
+function replaceRepeaterInFieldDef(fieldDef: LegendFieldDef<Field>, repeater: RepeaterValue): LegendFieldDef<string> | null {
   const field = fieldDef.field;
   if (isRepeatRef(field)) {
     if (field.repeat in repeater) {
-      return {
+      fieldDef = {
         ...fieldDef,
         field: repeater[field.repeat]
       };
     } else {
       log.warn(log.message.noSuchRepeatedValue(field.repeat));
+      // return null, meaning that the field def should be ignored
```

Comment:
Normally we use undefined for the rest of the code base.

Reply:
Done